.github/workflows/reusable-plugin-prep-release.yml

```diff
@@ -219,54 +219,96 @@ jobs:
             "[.[] | select(.mergedAt > \"$TAG_DATE\") | \"- #\(.number) \(.title)\"] | join(\"\\n\")")
 
           if [ -z "$MERGED_PRS" ]; then
-            echo "No pull requests merged since last release."
+            echo "No pull requests merged since last release. Skipping PR listing."
             echo "prs=No pull requests merged since last release to the ${PLUGIN_SOURCE_BRANCH} branch." >> "$GITHUB_OUTPUT"
             exit 0
           fi
 
           {
-            echo "prs=<<EOF"
+            echo "prs<<EOF"
             echo "$MERGED_PRS"
             echo "EOF"
           } >> "$GITHUB_OUTPUT"
 
-      # create the release branch, commit the changes, and create pull request
-      - name: Create Release Pull Request
-        id: create_pr
-        uses: peter-evans/create-pull-request@271a8d0340265f705b14b6d32b9829c1cb33d45e  # v7.0.8
-        with:
-          token: ${{ github.token }}
-          base: ${{ inputs.plugin-target-branch }}
-          branch: release/${{ steps.new.outputs.version }}
-          title: "Automated ${{ inputs.level }} release: ${{ steps.new.outputs.version }}"
-          body: |
-            ## Changes in this pull request
+      # create the release branch, commit the changes with the github-actions[bot]
+      - name: Create and commit to release branch
+        id: push-changes
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OLD_VERSION: ${ steps.old.outputs.version }}
+          NEW_VERSION: ${{ steps.new.outputs.version }}
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
 
-            - Bumped version from `${{ steps.old.outputs.version }}` to `${{ steps.new.outputs.version }}`
-            - Updated build files
-            - Updated language files
+          # Use GitHub token to authenticate the push
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
```

Comment:
I think if we change `persist-credentials` to `true`, we wouldn't need this authentication line.

Reply:
ok, you mean, change `persist-credentials` to true in the checkout step?

Reply:
I've updated, but wanted to confirm that's what you meant.

Reply:
Yes, correct! That should ensure we don't have to re-authenticate here.